exchange/exchange.go

```diff
@@ -590,3 +580,29 @@ func (e *exchange) makeBid(Bids []*pbsOrtbBid, adapter openrtb_ext.BidderName) (
 	}
 	return bids, errList
 }
+
+func buildResolvedRequest(bidRequest *openrtb.BidRequest) (json.RawMessage, error) {
+	// Snapshot of resolved bid request for debug if test request
+	var resolvedRequest json.RawMessage = nil
+	var err error
+
+	if bidRequest.Test == 1 {
+		if r, err := json.Marshal(bidRequest); err == nil {
+			resolvedRequest = r
+		}
+	}
+	return resolvedRequest, err
```

Comment:
I think this func should return an error if the call to `json.Marshal` fails, but that's not what happens. A new err variable is being declared on line 590 local to the `if` statement and the err defined on line 587 will never be set. Consider something like:

```
if bidRequest.Test == 1 {
    return json.Marshal(bidRequest);
}

return nil, nil
```

.. or something like that. Not sure if you need to do manual casting between `byte[]` and `json.RawMessage`.

Reply:
Pretty good catch. I don't need we need to cast to `[]byte`, do we? `resolvedRequest` ends up being passed as an argument to `buildBidResponse` which expects a `json.RawMessage` as it's fifth argument.
```
func (e *exchange) buildBidResponse(ctx context.Context, liveAdapters []openrtb_ext.BidderName, adapterBids map[openrtb_ext.BidderName]*pbsOrtbSeatBid, bidRequest *openrtb.BidRequest, resolvedRequest json.RawMessage, adapterExtra map[openrtb_ext.BidderName]*seatResponseExtra, errList []error) (*openrtb.BidResponse, error) {
```

Reply:
Done